File: app/src/main/java/com/wheat7/nationalgeographic/ui/activity/WebActivity.java
package com.wheat7.nationalgeographic.ui.activity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.wheat7.nationalgeographic.R;
import com.wheat7.nationalgeographic.databinding.ActivityWebBinding;

/**
 * Created by wheat7 on 2017/8/19.
 */

//WebView 使用kotlin会有问题，后续版本解决
public class WebActivity extends BaseActivity<ActivityWebBinding> {

    @Override
    public int getLayoutId() {
        return R.layout.activity_web;
    }

    @Override
    public void onBackPressed() {
        if (getBinding().webView.canGoBack()) {
            getBinding().webView.goBack();
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public void initView(Bundle savedInstanceState) {
        Intent intent = getIntent();
        String url = intent.getStringExtra("URL");
        getBinding().webView.getSettings().setJavaScriptEnabled(true);
        getBinding().webView.loadUrl(url);
        getBinding().webView.getSettings().setSupportZoom(true);
        getBinding().webView.getSettings().setDisplayZoomControls(false);
        getBinding().webView.getSettings().setUseWideViewPort(true);
        getBinding().webView.getSettings().setLoadWithOverviewMode(true);
        getBinding().webView.getSettings().setAppCacheEnabled(true);
        WebChromeClient chromeClient = new WebChromeClient() {
            @Override
            public void onReceivedTitle(WebView view, String title) {
                super.onReceivedTitle(view, title);
            }

            @Override
            public void onProgressChanged(WebView view, int newProgress) {
                super.onProgressChanged(view, newProgress);
                getBinding().webProgress.setProgress(newProgress);
            }
        };

        getBinding().webView.setWebChromeClient(chromeClient);
        getBinding().webView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
                getBinding().tvClose.setVisibility(View.VISIBLE);
                return super.shouldOverrideUrlLoading(view, request);
            }

            @Override
            public void onPageStarted(WebView view, String url, Bitmap favicon) {
                super.onPageStarted(view, url, favicon);
                getBinding().webProgress.setVisibility(View.VISIBLE);
            }

            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);
                getBinding().webProgress.setVisibility(View.GONE);
            }
        });

        getBinding().webBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });

        getBinding().tvClose.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                WebActivity.this.finish();
            }
        });
    }



}

File: app/src/main/java/com/wheat7/nationalgeographic/ui/activity/WebActivity.kt
package com.wheat7.nationalgeographic.ui.activity

import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.webkit.WebChromeClient
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import android.support.v7.widget.Toolbar;

import com.wheat7.nationalgeographic.R
import com.wheat7.nationalgeographic.databinding.ActivityWebBinding
import kotlinx.android.synthetic.main.activity_web.*

/**
 * Created by wheat7 on 2017/8/19.
 */

//因为LifecycleActivity中没有了setSupportActionBar()和getSupportActionBar(),未解决，所以不能继承BaseActitivy
class WebActivity : AppCompatActivity() {

    override fun onBackPressed() {
        if (web_view.canGoBack()) {
            web_view.goBack()
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web)
        val intent = intent

        val url = intent.getStringExtra("URL")
        web_view.settings.javaScriptEnabled = true
        web_view.loadUrl(url)
        web_view.settings.setSupportZoom(true)
        web_view.settings.displayZoomControls = false
        web_view.settings.useWideViewPort = true
        web_view.settings.loadWithOverviewMode = true
        web_view.settings.setAppCacheEnabled(true)
        setSupportActionBar(toolbar_web)
        getSupportActionBar()!!.setDisplayShowTitleEnabled(false)
        val chromeClient = object : WebChromeClient() {
            override fun onReceivedTitle(view: WebView, title: String) {
                super.onReceivedTitle(view, title)
                //getBinding().webTitle.setText(title)
            }

            override fun onProgressChanged(view: WebView, newProgress: Int) {
                super.onProgressChanged(view, newProgress)
                web_progress.progress = newProgress
            }
        }

        web_view.webChromeClient = chromeClient
        web_view.webViewClient = object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
                tv_close.setVisibility(View.VISIBLE)
                return super.shouldOverrideUrlLoading(view, request)
            }

            override fun onPageStarted(view: WebView, url: String, favicon: Bitmap) {
                super.onPageStarted(view, url, favicon)
                web_progress.visibility = View.VISIBLE
            }

            override fun onPageFinished(view: WebView, url: String) {
                super.onPageFinished(view, url)
                web_progress.visibility = View.GONE
            }
        }
    }




}
